test/fast/git-process-test.ts

```diff
@@ -2,15 +2,35 @@ import * as path from 'path'
 import * as fs from 'fs'
 import * as crypto from 'crypto'
 
-import { GitProcess, GitError, RepositoryDoesNotExistErrorCode } from '../../lib'
+import {
+  GitProcess,
+  GitError,
+  RepositoryDoesNotExistErrorCode,
+  GitTaskCancelResult
+} from '../../lib'
 import { GitErrorRegexes } from '../../lib/errors'
 import { initialize, verify, initializeWithRemote, gitForWindowsVersion } from '../helpers'
 
 import { gitVersion } from '../helpers'
+import { setupNoAuth } from '../slow/auth'
 
 const temp = require('temp').track()
 
 describe('git-process', () => {
+  it('clone-and-cancel', async () => {
+    fs.mkdtemp('desktop-git-clone-and-cancel', async (err, folder) => {
+      const options = {
+        env: setupNoAuth()
+      }
+      const result = GitProcess.execTask(
+        ['clone', '--', 'https://github.com/maifeeulasad/maifeeulasad.github.io', '.'],
```

Comment:
Sorry @maifeeulasad I didn't see this until now 😓  Have you tried this?
```suggestion
        ['clone', 'https://github.com/maifeeulasad/maifeeulasad.github.io', folder],
```